src/client.rs

```diff
@@ -313,8 +313,18 @@ impl RequestBuilder {
                         headers.set(Referer(url.to_string()));
                         urls.push(url);
                         let action = check_redirect(&client.redirect_policy.lock().unwrap(), &loc, &urls);
+
                         match action {
-                            redirect::Action::Follow => loc,
+                            redirect::Action::Follow => {
+                                let cross_host = loc.host().unwrap() != urls.last().unwrap().host().unwrap();
+                                if cross_host {
+                                    headers.remove::<Authorization<::header::Basic>>();
+                                    headers.remove::<Authorization<String>>();
+                                    headers.remove::<Authorization<::header::Bearer>>();
```

Comment:
@seanmonstar I tried with just `headers.remove::<Authorization>()` but I got this error - 

```

error[E0243]: wrong number of type arguments: expected 1, found 0
   --> src/client.rs:321:54
    |
321 |                                     headers.remove::<Authorization>();
    |                                                      ^^^^^^^^^^^^^ expected 1 type argument

error: aborting due to previous error
```
Then I checked `hyper`'s docs for the `Authorization` header here - https://hyper.rs/hyper/0.8.0/hyper/header/struct.Authorization.html and found that it needs a type param. So I wrote three lines for all three possible Authorization types. Is there some other way to solve this?